source/material_model/rheology/composite_viscous_creep.cc

```diff
@@ -0,0 +1,238 @@
+            strain_rate_deriv = edot_and_deriv.second;
+
+            // If the strain rate derivative is zero, we catch it below.
+            if (strain_rate_deriv>std::numeric_limits<double>::min())
+              creep_stress -= strain_rate_residual/strain_rate_deriv;
+            stress_iteration += 1;
+          }
+
+        // The creep stress is not the total stress, so we still need to do a little work to obtain the effective viscosity.
+        // First, we compute the stress running through the strain rate limiter
+        const double lim_stress = 2.*min_viscosities[composition]*(edot_ii - creep_stress/(2.*max_viscosities[composition]));
+
+        // Now we return the viscosity using the total stress
+        return (creep_stress + lim_stress)/(2.*edot_ii);
+      }
+
+
+
+      template <int dim>
+      std::pair<double, double>
+      CompositeViscousCreep<dim>::compute_strain_rate_and_derivative (const double creep_stress,
+                                                                      const double pressure,
+                                                                      const double temperature,
+                                                                      const DiffusionCreepParameters diffusion_creep_parameters,
+                                                                      const DislocationCreepParameters dislocation_creep_parameters,
+                                                                      const PeierlsCreepParameters peierls_creep_parameters,
+                                                                      const double min_viscosity,
+                                                                      const double max_viscosity) const
+      {
+        const std::pair<double, double> diff_edot_and_deriv = diffusion_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, diffusion_creep_parameters);
+        const std::pair<double, double> disl_edot_and_deriv = dislocation_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, dislocation_creep_parameters);
+        const std::pair<double, double> prls_edot_and_deriv = peierls_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, peierls_creep_parameters);
+
+        const double strain_rate = creep_stress/(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*(diff_edot_and_deriv.first + disl_edot_and_deriv.first + prls_edot_and_deriv.first);
+        const double strain_rate_deriv = 1./(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*(diff_edot_and_deriv.second + disl_edot_and_deriv.second + prls_edot_and_deriv.second);
+
+        return std::make_pair(strain_rate, strain_rate_deriv);
+      }
+
+
+
+      template <int dim>
+      void
+      CompositeViscousCreep<dim>::declare_parameters (ParameterHandler &prm)
+      {
+
+        prm.declare_entry ("Minimum strain rate", "1.4e-20", Patterns::Double(0.),
```

Comment:
There is already a parameter called Minimum strain rate in the `visco_plastic` (and `diffusion_dislocation`?) material model. To avoid confusion, should this be called the `Minimum creep strain rate`? Perhaps a minimum strain rate should be defined not in the specific material model (or rheology) subjection, but rather as a main material model parameter? It seems like the value specified here and the one used by the material model should be consistent.

Reply:
I think this parameter should be shared by both the material model and the rheology model. This is how the current code does it (because we don't enter different subsections for the rheology models). Because this material model uses Minimum strain rate, it should be declared here, but it can share the parameter given to it by the user with the rheology model.

I've added a comment saying that this parameter is shared with the rheology models.